sdk/python/kfp/registry/registry_client.py

```diff
@@ -152,73 +157,139 @@ def _validate_tag(self, tag: str) -> None:
         if tag.startswith(_VERSION_PREFIX):
             raise ValueError('Tag should not start with \"sha256:\".')
 
-    def load_config(self) -> dict:
+    def load_auth(
+        self,
+        auth: Optional[Union[requests.auth.AuthBase,
+                             credentials.Credentials]] = None,
+        auth_file: Optional[str] = None
+    ) -> Optional[Union[requests.auth.AuthBase, credentials.Credentials]]:
+        """Loads the credentials for authentication.
+
+        Args:
+            auth: Optional. Authentication using python requests or google.auth credentials.
+            auth_file: Optional. The location of the local config file that contains the
+                authentication token. If not specified, defaults to
+                LOCAL_REGISTRY_CREDENTIAL (if it exists).
+
+        Returns:
+            The loaded authentication token.
+        """
+        if auth:
+            return auth
+        elif self._is_ar_host():
+            auth, _ = google.auth.default()
+            return auth
+        elif auth_file:
+            if os.path.exists(auth_file):
+                # Fetch auth token using the locally stored credentials.
+                with open(auth_file, 'r') as f:
+                    auth_token = json.load(f)
+                    return ApiAuth(auth_token)
+            else:
+                raise ValueError(f'Auth file not found: {auth_file}.')
+        elif os.path.exists(LOCAL_REGISTRY_CREDENTIAL):
+            # Fetch auth token using the locally stored credentials.
+            with open(LOCAL_REGISTRY_CREDENTIAL, 'r') as f:
+                auth_token = json.load(f)
+                return ApiAuth(auth_token)
+        return None
+
+    def load_config(self, host: str, config_file: str) -> dict:
         """Loads the config.
 
+        Args:
+            host: The address of the registry host. The host needs to be specified here
+                or in the config file.
+            auth: Optional. Authentication using python requests or google.auth credentials.
+            config_file: Optional. The location of the local config file. If not specified,
+                defaults to LOCAL_REGISTRY_CONTEXT (if it exists).
+            auth_file: Optional. The location of the local config file. If not specified,
+                defaults to LOCAL_REGISTRY_CREDENTIAL (if it exists).
+
         Returns:
-            The loaded config if it's a known host, otherwise None.
+            The loaded config.
         """
-        # TODO: Move config file code to its own file.
         config = {}
```

Comment:
nit: Consider moving this to an `else` clause on line 247 to make the purpose of its assignment more clear

Reply:
Done.